packages/migrate-config/src/migrate-config.js

```diff
@@ -0,0 +1,872 @@
+					"init",
+					b.identifier(name),
+					b.literal(globals[name]),
+				);
+			}),
+		);
+	}
+
+	return b.objectExpression(properties);
+}
+
+/**
+ * Creates the linter options object from the config.
+ * @param {Config} config The config to create linter options from.
+ * @returns {ObjectExpression|undefined} The linter options object or undefined if none.
+ */
+function createLinterOptions(config) {
+	if (!config.noInlineConfig && !config.reportUnusedDisableDirectives) {
+		return undefined;
+	}
+
+	const properties = [];
+
+	linterOptionsKeysToCopy.forEach(key => {
+		if (config[key]) {
+			properties.push(
+				b.property("init", b.identifier(key), b.literal(config[key])),
+			);
+		}
+	});
+
+	return b.objectExpression(properties);
+}
+
+/**
+ * Creates an array of function arguments from an array of extended configs.
+ * @param {string|string[]} extendedConfigs The extended configs to convert.
+ * @returns {Array<Literal>} The AST for the array expression.
+ */
+function createExtendsArguments(extendedConfigs) {
+	// create an array of strings
+	if (typeof extendedConfigs === "string") {
+		return [b.literal(extendedConfigs)];
+	}
+
+	return extendedConfigs.map(config => b.literal(config));
+}
+
+/**
+ * Creates a an object expression that duplicates an existing object.
+ * @param {Object} value The object to create the AST for.
+ * @returns {ObjectExpression|ArrayExpression|Literal} The AST for the object.
+ */
+function createAST(value) {
+	if (Array.isArray(value)) {
+		return b.arrayExpression(value.map(item => createAST(item)));
+	}
+
+	if (value && typeof value === "object") {
+		const properties = Object.keys(value).map(key => {
+			const propertyValue = value[key];
+			const identifier = isValidIdentifier(key)
+				? b.identifier(key)
+				: b.literal(key);
+			return b.property("init", identifier, createAST(propertyValue));
+		});
+
+		return b.objectExpression(properties);
+	}
+
+	return b.literal(value);
+}
+
+/**
+ * Creates an array expression from an array of glob patterns.
+ * @param {string[]} patterns The glob patterns to convert.
+ * @returns {ArrayExpression} The AST for the array expression.
+ */
+function createFilesArray(patterns) {
+	return b.arrayExpression(
+		patterns.map(pattern => b.literal(convertGlobPattern(pattern))),
+	);
+}
+
+/**
+ * Creates an object expression for the language options.
+ * @param {Migration} migration The migration object.
+ * @param {Config} config The config to create language options from.
+ * @returns {ObjectExpression|undefined} The AST for the object expression or undefined if none.
+ */
+function createLanguageOptions(migration, config) {
+	const properties = [];
+	const { imports, messages } = migration;
+
+	// Both `env` and `globals` end up as globals in flat config
+	const globals = createGlobals(config);
+	if (globals) {
+		properties.push(b.property("init", b.identifier("globals"), globals));
+	}
+
+	// For `env` we will need to the `globals` package
+	if (config.env) {
```

Comment:
```suggestion
	if (config.env && Object.keys(config.env).some(name => !name.startsWith("es"))) {
```

No need for the `globals` package if all environments are es*.